src/components/Storage/common/useMultiselect.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+
+function useSelected<T>(options: T[]): [Set<T>, (t: Set<T>) => void] {
+  const [selectedValue, setSelected] = useState<Set<T>>(new Set());
+
+  return [
+    new Set(options.filter((path) => selectedValue.has(path))),
+    setSelected,
+  ];
+}
+
+// Takes a list of elements and allow to select multiple like in gmail.
+export function useMultiselect<T>(options: T[]) {
+  const [selected, setSelected] = useSelected(options);
+  const [stack, setStack] = useState<T[]>([]);
```

Comment:
I'm not sure if I completely understood the "stack" concept here, so please let me know if I'm mistaken in the comments below. Would appreciate some comments on just storing the last clicked item isn't enough -- and may be a test case too.